dotfiles.sh

```diff
@@ -39,8 +40,13 @@ if [ $version = yes ]; then
     echo "$versionstr"
     exit 0
 fi
+if [ $edit = yes ]; then
+    cd $HOME/.dotfiles
+    exec ${EDITOR:-vi} ${*:-.}
+    exit 0
+fi
 if [ $# -ne 1 ]; then
-    echo "dotfiles: ERROR: invalid number of arguments" 1>&2
+    echo "dotfiles: ERROR: invalid number of arguments $#" 1>&2
```

Comment:
It's OK to print the number of actually provided arguments, but (a) it doesn't read nicely when you see "invalid number of arguments 7" and (b) the actual number of expected arguments could be more useful in the output. Perhaps a better error could be "invalid number of arguments (expected 1 argument, received N arguments)".